server/db/create_test_data.js

```diff
@@ -58,45 +72,43 @@ function createTestUser(recordCount, cb) {
     bulkData.push(generateTestRecord(j));
   }
 
-  db.createUser(fakeUser.id, fakeUser.email, fakeUser.oauthToken, function(err) {
-    if (err) { throw err; }
-    log.verbose('inserted fake user into db');
-
-    // instead of db.createVisit, we'll just issue a bulk insert query directly
-    var query = 'INSERT INTO visits (id, visitedAt, fxaId, rawUrl, url, urlHash, title) ' +
-                'VALUES ?';
-    log.trace('about to insert bulk data');
-    log.trace('query is: ' + query);
-    log.trace('bulkData.length is ' + bulkData.length);
-    pool.query(query, [bulkData], function(err) {
-      if (err) { throw err; }
-      log.verbose('inserted ' + bulkData.length + ' user visits into db');
-      cb && cb(err);
-    });
-  });
-}
-
-// TODO replace this with a create_db call, since it drops the old DB along the way
-function truncateTables(cb) {
-  pool.getConnection(function(err, c) {
+  user.create(fakeUser.id, fakeUser.email, fakeUser.oauthToken, function(err) {
     if (err) {
-      log.verbose('error getting connection: ' + err);
-      return cb(err);
-    }
-    c.query('TRUNCATE visits', function(err) {
+      return cb && cb(err);
+    } 
+    pg.connect(dbParams, function(err, client, done) {
       if (err) {
-        log.verbose('error truncating table: ' + err);
-      } else {
-        log.verbose('truncated visits table');
+        log.warn('failed to connect to database: ' + err);
+        done();
+        pg.end();
+        return cb && cb(err);
+      }
+
+      // instead of db.createVisit, we'll just issue a bulk insert query directly
+      var query = 'INSERT INTO visits (id, visitedAt, fxaId, rawUrl, url, urlHash, title) VALUES ';
+      // a micro query builder. turns arrays into quoted items enclosed in parens, comma-separated.
+      // for example, [foo, 1], [bar, 2] => "('foo', '1'), ('bar', '2')"
+      /*jshint quotmark: false */
+      while(bulkData.length) {
+        var next = bulkData.shift();
+        query += '(' + next.map(function(item) { return "'" + item + "'"; }).join(', ') + ')';
+        if (bulkData.length) { query += ', '; }
       }
-      c.query('TRUNCATE users', function(err) {
+      log.trace('about to insert bulk data');
+      log.trace('query is: ' + query);
+      log.trace('bulkData.length is ' + bulkData.length);
+      client.query(query, function(err, r) {
         if (err) {
-          log.verbose('error truncating tables: ' + err);
-        } else {
-          log.verbose('truncated users table');
+          log.warn('failed to create test data: ' + err);
+          done();
+          pg.end();
+          return cb && cb(err);
         }
-        c.release();
+        // TODO examine r and get the number of affected rows
+        log.verbose('inserted ' + recordCount + ' user visits into db');
         cb && cb(err);
+        done();
+        pg.end();
```

Comment:
Since we're always calling `done()` and `pg.end()`, does it make sense to move those into the callback? Or doesn't that make sense since we seem to not always be guaranteed to have a callback.
Conversely, if no callback function is specified, create a dummy function which calls `pg.end()`.